src/controller/AutoCommissioner.cpp

```diff
@@ -502,9 +502,41 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
             SetDAC(report.Get<RequestedCertificate>().certificate);
             break;
         case CommissioningStage::kSendAttestationRequest:
-            // These don't need to be deep copied to local memory because they are used in this one step then never again.
-            mParams.SetAttestationElements(report.Get<AttestationResponse>().attestationElements)
-                .SetAttestationSignature(report.Get<AttestationResponse>().signature);
+            if (report.Get<AttestationResponse>().attestationElements.size() > kAttestationElementsLength)
```

Comment:
```suggestion
            if (report.Get<AttestationResponse>().attestationElements.size() > sizeof(mAttestationElements))
```
since that's what the code guarded by this `if` really cares about, right?

Reply:
Also, this code would be a _lot_ more readable with something like:
```
auto & elements = report.Get<AttestationResponse>().attestationElements;
```
or so....

Reply:
addressed